sound/soc/sof/ipc4-topology.c

```diff
@@ -1357,18 +1358,24 @@ static int snd_sof_get_nhlt_endpoint_data(struct snd_sof_dev *sdev, struct snd_s
 						   &bit_depth);
 		if (ret < 0)
 			return ret;
+
+		/* for SSP, dev type could be 0 (BT sideband) or 4 (analog codec) */
```

Comment:
I would expand this comment and make it clear that the SSP connection is static and the NHLT have board specific configuration which describes the endpoint dev_type.
The dev_type is not specified in tplg, just the SSP  port and this is why we query the dev_type for the port and pass that back to the blob lookup function.
Iow, in SOF we are using the SSP port to find the blob and not the dev_type.

or something around these lines

Reply:
comment updated, please check